### PR TITLE
dts: stm32: added SPI 2&3 for stm32f105

### PR DESCRIPTION
**Pull request by mkolchurin**
Added device tree definition for SPI2 and SPI3 in STM32F105

Signed-off-by: Maxim Kolchurin maxim.kolchurin@gmail.com

### CODE REVIEW

File: dts/arm/st/f1/stm32f105.dtsi [hunk 1]
```diff
@@ -79,6 +79,28 @@
 			label = "UART_5";
 		};
 
+		spi2: spi@40038000 {
+			compatible = "st,stm32-spi";
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0x40038000 0x400>;
+			clocks = <&rcc STM32_CLOCK_BUS_APB2 0x00001000>;
```
Comment (FRASTM) [inline dts/arm/st/f1/stm32f105.dtsi:87]: SPI2 is on RCC_APB1ENR   bit 14   (according to the RefMan RM0008)

**Review by FRASTM — CHANGES_REQUESTED** [review]

File: dts/arm/st/f1/stm32f105.dtsi [hunk 2]
```diff
@@ -79,6 +79,28 @@
...
+			interrupts = <36 5>;
+			status = "disabled";
+			label = "SPI_2";
+		};
+
+		spi3: spi@4003C000 {
```
Comment (FRASTM) [inline dts/arm/st/f1/stm32f105.dtsi:93]: better is lower case '4003c000' for node and reg address

**Comment by erwango** [comment]
@mkolchurin Do you plan to fix update this PR?

**Review by erwango — DISMISSED** [review]

**Review by FRASTM — DISMISSED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Comment by mbolivar-nordic** [comment]
Could someone from ST please re-ack?

**Review by erwango — APPROVED** [review]

**Pull request merged**